### ISSUE TITLE
How can I run Lens with 'gimme-aws-creds'?

### ISSUE THREAD

**Issue by pranav-bhatt** [kontena/lens#6958]
Hi!

In my setup, I need to acquire temporary AWS credentials using `gimme-aws-creds` before accessing my clusters. Without these temporary credentials, I get the following error:
```
could not get token: NoCredentialProviders: no valid providers in chain. Deprecated.
	For verbose messaging see aws.Config.CredentialsChainVerboseErrors

Unable to connect to the server: getting credentials: exec: executable aws-iam-authenticator failed with exit code 1
```

I run an M1 mac, and when I launch the lens application, these credentials haven't got generated and stored as local env variables. This causes the above message to pop up for me, and I'm therefore not able to connect to my clusters.

Is it possible to somehow run `gimme-aws-config` in the same context as Lens? I noticed there was a path that could be provided for a terminal. If I put a script over there which runs `gimme-aws-config` before hand, will that work?

Thanks!

**Comment by Nokel81** [comment]
Why not run it as part of the `exec` field in your `kubeconfig`'s `user`?

**Comment by pranav-bhatt** [comment]
So the tool requires user input for verification purposes. Not sure if adding it to the exec field would allow me to input details.

**Comment by Nokel81** [comment]
Ah I see. Can you call it yourself and then run `Lens` from the terminal?

**Comment by pranav-bhatt** [comment]
Hey, sorry for the late reply. I've been trying to install the Lens CLI, but the install button doesn't seem to do anything after prompting for the password (I'm running macOS 12.6)

**Comment by Nokel81** [comment]
No I mean running something like `/Applications/Lens.app/Contents/MacOS/Lens`

**Comment by pranav-bhatt** [comment]
Yeah I tried with that, and ran it with sudo as well, but no luck. Still got the authentication error.

**Comment by dennisshpits** [comment]
Lens should work with gimme-aws-creds already. When you run gimme-aws-creds it should generate a credentials file that is under the .aws folder. You might need to set this up. Once you do that you might have to configure your kube config files to look at the credentials file. Anyway this isn’t an issue with Lens. Good luck!
